omod/src/main/webapp/resources/scripts/components/LabTrackingOrderFactory.js

```diff
@@ -390,12 +405,49 @@ angular.module("labTrackingOrderFactory", [])
             _updateObsIfExists(labTrackingOrder, "accessionNumber", obs, obsIdsToDelete);
             _updateObsIfExists(labTrackingOrder, "urgentReview", obs, obsIdsToDelete);
 
-            if (labTrackingOrder.postopDiagnosis.diagnosis.value != null && labTrackingOrder.postopDiagnosis.diagnosis.value.length > 0) {
-
+            // special case for post-op diagnosis obs group
+            if (((typeof labTrackingOrder.postopDiagnosis.diagnosis === 'object' && labTrackingOrder.postopDiagnosis.diagnosis.value === null)
+                || labTrackingOrder.postopDiagnosis.diagnosis === null || labTrackingOrder.postopDiagnosis.diagnosis.length === 0)) {
```

Comment:
Again the fact that `null` is an object will create trouble. For the post op stuff you definitely have `type`, right? Would be more reliable and clearer to just use that.

Also you can just use a nullish check for the string/null value.

```suggestion
            if ((labTrackingOrder.postopDiagnosis.type === 'coded' && labTrackingOrder.postopDiagnosis.diagnosis.value === null)
                || !labTrackingOrder.postopDiagnosis.diagnosis) {
```

Reply:
I'll explain the "type" stuff below, but the "null" point is valid... I should explicitly check for null (and make sure it does the right thing)

Reply:
added a null check (and also some more comments, as this is a confusing block)